cetmix_tower_server/models/cx_tower_server.py

```diff
@@ -403,20 +403,27 @@ def _compute_file_count(self):
 
     @api.constrains("ip_v4_address", "ip_v6_address", "ssh_auth_mode")
     def _constraint_ssh_settings(self):
-        """Ensure SSH settings are valid"""
-        for rec in self:
-            if not rec.ip_v4_address and not rec.ip_v6_address:
-                raise ValidationError(
-                    _("Please provide IPv4 or IPv6 address for %(srv)s", srv=rec.name)
-                )
-            if rec.ssh_auth_mode == "p" and not rec.ssh_password:
-                raise ValidationError(
-                    _("Please provide SSH password for %(srv)s", srv=rec.name)
-                )
-            if rec.ssh_auth_mode == "k" and not rec.ssh_key_id:
-                raise ValidationError(
-                    _("Please provide SSH Key for %(srv)s", srv=rec.name)
-                )
+        """Ensure SSH settings are valid.
+        Set 'skip_ssh_settings_check' context key to skip the checks
+        """
+
+        if not self._context.get("skip_ssh_settings_check"):
```

Comment:
in order to reduce code nesting you can make a positive check:

```python
if self._context.get("skip_ssh_settings_check"):
    return False

for rec in self:
    ...
```

Reply:
@GabbasovDinar I thought about it however this is a constrain function that doesn't return anything. So I decided to go with a negative check.